src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -257,12 +257,42 @@ def getReplicaInfoForBlocks(self, **kwargs):
 
         return result
 
-    def getPFN(self, nodes=None, lfns=None, destination=None, protocol='srmv2', custodial='n'):
-        """
-        Copy of the method from the PhEDEx service class
-        Used by CRABServer
-        """
-        raise NotImplementedError("Apparently not available in the Rucio client as well")
+    def getPFN(self, site, lfns, protocol=None):
+        """
+        returns a lost of PFN(s) for a list of LFN(s) and one site
+        Note: same function implemented for PhEDEx accepted a list of sites, but semantic was obscure and actual need
+              unknown. So take this chance to make things simple.
+        :param site: a Rucio RSE, i.e. a site name in standard CMS format like 'T1_UK_RAL_Disk' or  'T2_CH_CERN'
+        :param lfns: a list of LFN's does not need to correspond to actual files and could be a top level directory
+                      like ['/store/user/rucio/jdoe','/store',...] basically any string starting with '/' is
+                      accepted and LFN -> PFN translation is almost always a simple prefix
+        :param protocol: If the RSE supports multiple access protocols, a preferred protocol can be selected via this,
+                         otherwise the default one for the site will be selected. Example: 'gsiftp' or 'davs'
+        :return: a dictionary having the LFN's as keys and the corresponding PFN's as values.
+
+        Will raise a Rucio exception is input is wrong.
+        """
+
+        # allow for duck typing, if a single lfn was passed, make it a list
+        # avoid handling a string like a list of chars !
+        lfns = [lfns] if isinstance(lfns, basestring) else lfns
+
+        # add a scope to turn LFNs into Rucio DID syntax
+        dids = ['cms:'+lfn for lfn in lfns]
+
+        # Rucio's lfns2pfns returns a dictionary with did as key and pfn as value:
+        # {u'cms:/store/user/rucio': u'gsiftp://red-gridftp.unl.edu:2811/mnt/hadoop/user/uscms01/pnfs/unl.edu/data4/cms/store/user/rucio'}
+        didDict = self.cli.lfns2pfns(site, dids, scheme=protocol)
+
+        # convert to a more useful format for us with LFN's as keys
+        # there's surely a more pythonic way to do this, py3 compatible etc. but since this will almost only
+        # be called with a single LFN, not a list.. why bother ? I just make it clear (and stupid)
+        pfnDict = {}
+        for oldKey in didDict.keys():
```

Comment:
```suggestion
        for oldKey in didDict:
```
this will make it compatible with py2 and py3